src/transformers/generation/configuration_utils.py

```diff
@@ -355,7 +357,7 @@ def __init__(self, **kwargs):
         self.sequence_bias = kwargs.pop("sequence_bias", None)
         self.guidance_scale = kwargs.pop("guidance_scale", None)
         self.low_memory = kwargs.pop("low_memory", None)
-        self.watermarking_args = kwargs.pop("watermarking_args", None)
+        self.watermarking_config = WatermarkingConfig.from_dict(kwargs.pop("watermarking_config", None))
```

Comment:
```suggestion
        watermarking_config = kwargs.pop("watermarking_config", None)
        if watermarking_config is None:
            self.watermarking_config = None
        elif isinstance(watermarking_config, WatermarkingConfig):
            self.watermarking_config = watermarking_config
        else:
            self.watermarking_config = WatermarkingConfig.from_dict(watermarking_config)
```

1.`WatermarkingConfig.from_dict()` should complain if it gets a `None` input, like the other config classes
2. No transformation should be done if the input is already a config